storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -233,21 +245,16 @@ lsn_t Redo_Log_Reader::read_log_seg_8030(log_t &log, byte *buf, lsn_t start_lsn,
   ut_a(start_lsn < end_lsn);
 
   // update the in-memory structure log files by scanning
-  if (log.m_files.find(start_lsn) == log.m_files.end()) {
-    if (!reopen_log_files(start_lsn)) {
-      // file not found
-      m_error = true;
-      return 0;
-    }
-  }
-
-  auto file = log.m_files.find(start_lsn);
-  if (file == log.m_files.end()) {
-    // file not found
+  if (log.m_files.find(start_lsn) == log.m_files.end() &&
+      !reopen_log_files(start_lsn)) {
     m_error = true;
     return 0;
   }
 
+  auto file = log.m_files.find(start_lsn);
+
+  ut_ad(file != log.m_files.end());
```

Comment:
change it to a graceful exit. We should avoid ut on xtrabackup.

Reply:
PXB should find file from reopen. If there is an issue reopen would exit. So that's why added the assert just an additional safety.

Reply:
it's a debug assertion, so it's okay. In general, pxb should always handle failures gracefully with a meaningful error message that tells the user exactly what went wrong.